File: Examples2018/UIViewPager/app/src/main/java/course/examples/ui/viewpager/GalleryWithViewPagerActivity.java
// This project requires the v13 support library. 
// See http://developer.android.com/tools/support-library/setup.html for more information


package course.examples.ui.viewpager;

import android.app.Activity;
import android.os.Bundle;
import android.support.v4.view.ViewPager;

public class GalleryWithViewPagerActivity extends Activity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.main);

		ViewPager viewPager = findViewById(R.id.pager);

		// Create a new ImageAdapter (subclass of FragmentStatePagerAdapter)
		// ViewPager uses support library, so use getSupportFragmentManager()
		// instead of getFragmentManager()
		ImageAdapter imageAdapter = new ImageAdapter(getFragmentManager());

		// Set the Adapter on the ViewPager
		viewPager.setAdapter(imageAdapter);

	}
}

File: ExamplesKotlin/UIViewPager/app/src/main/java/course/examples/ui/viewpager/GalleryWithViewPagerActivity.kt
// This project uses the v13 support library.
// See http://developer.android.com/tools/support-library/setup.html for more information


package course.examples.ui.viewpager

import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.support.v4.view.ViewPager

class GalleryWithViewPagerActivity : FragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        val viewPager = findViewById<ViewPager>(R.id.pager)

        // Create a new ImageAdapter (subclass of FragmentStatePagerAdapter)
        val imageAdapter = ImageAdapter(getImageIds(), supportFragmentManager)

        // Set the Adapter on the ViewPager
        viewPager.adapter = imageAdapter

    }

    private fun getImageIds(): IntArray {

       val resIds = resources.obtainTypedArray(R.array.image_ids)

        val ids = IntArray(resIds.length())
        for (i in ids.indices) {
            ids[i] = resIds.getResourceId(i, 0)
        }

        resIds.recycle()

        return ids
    }
}
